README.md

```diff
@@ -1,289 +1,610 @@
 
+If a `.spec.type` is set, the `type` entry in the binding `Secret` **MUST** be set to its value. If a `.spec.provider` is set, the `provider` entry in the binding `Secret` **MUST** be set to its value.
 
-#### Exposing data as environment variables
+A Synthetic Provisioned Service resource **MUST** define a `.status.binding.name` which is a `LocalObjectReference` to a `Secret`. The `Secret` **MUST** be in the same namespace as the resource. The `Secret` **MUST** contain a `type` entry with a value that identifies the abstract classification of the binding. It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding. The `Secret` **MAY** contain any other entry.
 
-The specification allows for binding properties to be additionally exposed as environment variables.  The assumption is that applications will have pre-knowledge of these environment variable keys, so the mechanisms described in this section are solely for the purpose of allowing the entity responsible for mounting the binding data to know which keys should also be made available as environment variables.  
+### Resource Type Schema
 
-This decision is made in the following ascending order of precedence:
-* value of `ServiceBinding`'s global `bindAs` element, which applies to all binding data.
-* value of the `bindAs` attribute in each of the `dataMappings` elements inside `ServiceBinding`.
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: SyntheticProvisionedService
+metadata:
+  name:         # string
+spec:
+  type:         # string, optional
+  provider:     # string, optional
+
+  services:     # []ObjectReference
+  - apiVersion: # string
+    kind:       # string
+    name:       # string
+    ...
+
+  mappings:     # []Mapping, optional
+  - name:       # string
+    value:      # string
+
+status:         # Provisioned Service Duck Type
+  binding:
+    name:       # string
+```
+
+### Example Resource
+
+```yaml
+apiVersion: service.binding/v1alpha1
+kind: SyntheticProvisionedService
+metadata:
+  name: kafka-composite-service
+spec:
+  name: kafka
+  type: Kafka
+
+  services:
+  - apiVersion: event.stream/v1beta1
+    kind:       Cluster
+    name:       prod-kafka-cluster
+  - apiVersion: event.stream/v1beta1
+    kind:       User
+    name:       prod-kafka-user
+
+  mappings:
+  - name: event-streams-url
+    value: ((prod-kafka-cluster.status.url))/?username=((prod-kafka-user.status.username))
```

Comment:
@nebhale - in the `mappings` of a `ServiceBinding` CR the `value` referenced things from the service's `Secret`.  Here it seems we are referencing items from the actual service CR.  I think to be consistent, the example here should be be:

```
value: ((prod-kafka-cluster.url))/?username=((prod-kafka-user.username))
```

In other words:  just reference the items from the processed `Secret` of the particular services.

Reply:
Resolved by f5af146